include/lsst/meas/extensions/trailedSources/VeresModel.h

```diff
@@ -0,0 +1,100 @@
+// -*- lsst-c++ -*-
+/*
+ * This file is part of meas_extensions_trailedSources.
+ *
+ * Developed for the LSST Data Management System.
+ * This product includes software developed by the LSST Project
+ * (https://www.lsst.org).
+ * See the COPYRIGHT file at the top-level directory of this distribution
+ * for details of code ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef LSST_MEAS_EXTENSIONS_TRAILEDSOURCES_VERES_H
+#define LSST_MEAS_EXTENSIONS_TRAILEDSOURCES_VERES_H
+
+#include "lsst/pex/config.h"
+#include "lsst/afw/image.h"
+#include "lsst/afw/geom.h"
+#include "lsst/geom.h"
+#include "ndarray.h"
+
+namespace lsst {
+namespace meas {
+namespace extensions {
+namespace trailedSources {
+
+/**
+ * Implementation of an axisymmetric 2D Gaussian convolved with a line -- a model
+ * for a fast-moving, trailed-source (Veres et al. 2012).
+ *
+ * VeresModel is designed to compute the chi-squared of the model given an
+ * `lsst::afw::image::Exposure`, and to be passed to scipy.optimize for minimization.
+ */
+class VeresModel {
+public:
+    typedef afw::image::Image<float> Image;
+    typedef afw::image::Exposure<float> Exposure;
+
+    /**
+     * Constructor for VeresModel.
+     *
+     * @param data Exposure passed from the measurement task.
+     */
+    explicit VeresModel(Exposure const& data);
+
```

Comment:
Ok. I've added these and everything compiles, but I'm not entirely sure how to think about/check the behavior of the default constructors...

Reply:
The issue was that a `ImageF::Array` (and LSST image-ish types in general) are designed for shallow copies, so the compiler-generated copy-constructor would create two `VeresModel` objects with shared state. Since I see the only remaining members are `_data` and `_variance`, which are supposed to be fixed at construction time, this is probably ok now.

Reply:
Ah ok, so the (now removed) `_model` would've been updated inside of both the original and the 'copied' class? And to be sure, with that resolved, the default constructors (in the link above) will suffice?

Reply:
They look reasonable to me, but ultimately it's up to you to decide what is the correct behavior for your class.